pytorch_lightning/trainer/callback_hook.py

```diff
@@ -257,18 +257,21 @@ def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> Dict[Type, dict]:
             else:
                 state = callback.on_save_checkpoint(self, self.lightning_module, checkpoint)
             if state:
-                callback_states[type(callback)] = state
+                callback_states[callback.state_identifier] = state
         return callback_states
 
-    def on_load_checkpoint(self, checkpoint):
+    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
         """Called when loading a model checkpoint."""
         callback_states = checkpoint.get('callbacks')
```

Comment:
Let's annotate this for clarity

```suggestion
        callback_states: Dict[str, dict] = checkpoint.get('callbacks')
```